crates/sats/src/de.rs

```diff
@@ -551,6 +551,9 @@ pub use spacetimedb_bindings_macro::Deserialize;
 ///
 /// The lifetime `'de` allows us to deserialize lifetime-generic types in a zero-copy fashion.
 ///
+/// Do not manually implement this trait unless you know what you are doing.
+/// Incorrect implementations are safe, but can result in data loss.
```

Comment:
I would prefer a more precise warning that implementations of this trait must be consistent with all of `Serialize for T`, `SpacetimeType for T` and `Serialize, Deserialize for AlgebraicValue`. That should be plenty to scare off anyone who doesn't know what they're doing.